pallets/xcm-core-buyer/src/lib.rs

```diff
@@ -0,0 +1,456 @@
+                return Err(Error::<T>::OrderAlreadyExists.into());
+            }
+            in_flight_orders
+                .try_insert(para_id)
+                .map_err(|_| Error::<T>::InFlightLimitReached)?;
+
+            // Check that the para id is a parathread
+            let parathread_params = T::GetParathreadParams::get_parathread_params(para_id);
+            if parathread_params.is_none() {
+                return Err(Error::<T>::NotAParathread.into());
+            }
+
+            // TODO: also compare the latest slot from pallet_author_noting with parathread_params.slot_frequency
+
+            let xcm_weights_storage =
+                XcmWeights::<T>::get().ok_or(Error::<T>::XcmWeightStorageNotSet)?;
+
+            let withdraw_amount = xcm_weights_storage.buy_execution_cost;
+
+            // Send xcm to the relay
+            // Use a derivative account from the sovereign account based on the paraId
+            // Buy on-demand cores
+            // Any failure should return everything to the derivative account
+
+            // Don't use utility::as_derivative because that will make the tanssi sovereign account
+            // pay for fees, instead use `DescendOrigin` to make the parathread tank account
+            // pay for fees. The parathread tank account is derived from the tanssi sovereign
+            // account and the parathread para id.
+            // TODO: when coretime is implemented, buy core there instead of buying on-demand cores
+            let origin = OriginKind::SovereignAccount;
+            // TODO: max_amount is the max price of a core that this parathread is willing to pay
+            // It should be defined in a storage item somewhere, controllable by the container chain
+            // manager.
+            let max_amount = u128::MAX;
+            let call = T::GetPurchaseCoreCall::get_encoded(max_amount, para_id);
+            let weight_at_most = xcm_weights_storage.weight_at_most;
+
+            // Assumption: derived account already has DOT
+            // The balance should be enough to cover the `Withdraw` needed to `BuyExecution`, plus
+            // the price of the core, which can change based on demand.
+            let relay_asset_total: MultiAsset = (Here, withdraw_amount).into();
+            let refund_asset_filter: MultiAssetFilter =
+                MultiAssetFilter::Wild(WildMultiAsset::AllCounted(1));
+            // TODO: need better names for this methods.
+            //  interior_multilocation is the one used in DescendOrigin
+            //  absolute_multilocation is the one used in DepositAsset
+            // They can be easily converted from one another, the difference is that absolute_multilocation
+            // has an extra "Parachain" junction in the front, using SelfParaId::get()
+            let interior_multilocation = Self::interior_multilocation(para_id);
+            let derived_account = Self::absolute_multilocation(interior_multilocation);
```

Comment:
```suggestion
            let derived_account = Self::relay_relative_multilocation(interior_multilocation);
```

Reply:
We could also rename `interior_multilocation` to something like `local_relative_multilocation` or `local_interior_multilocation`, to emphasize on the fact that the point of view of that multilocation is the local origin.